internal/modules/distribution/test.py

```diff
@@ -0,0 +1,197 @@
+        rewards = query_rewards_singleval(delegator1, validator1)[1]["rewards"][0][
+            "amount"
+        ]
+
+        # withdraw rewards tx
+        status, rewards_tx = tx_withdraw_rewards("account1", validator1)
+        self.assertTrue(status)
+        self.assertTrue(rewards_tx)
+        time.sleep(3)
+
+        # query balance
+        after_balance = query_balances(delegator1)[1]["balances"][0]["amount"]
+
+        self.assertEqual(
+            (int(before_balance) + int(float(rewards))), int(after_balance)
+        )
+        time.sleep(5)
+
+    def test_withdraw_all_rewards_tx(self):
+        # query balance
+        before_balance = query_balances(delegator1)[1]["balances"][0]["amount"]
+        # query all rewards
+        all_rewards = query_rewards(delegator1)[1]["rewards"]
+        total = 0
+        for x in all_rewards:
+            rewards = x["reward"][0]["amount"]
+            total = total + (float(rewards))
+
+        # tx withdraw all rewards
+        status, all_rewards = tx_withdraw_allrewards("account1")
+        self.assertTrue(status)
+        self.assertTrue(all_rewards)
+
+        after_balance = query_balances(delegator1)[1]["balances"][0]["amount"]
+        self.assertEqual(
+            (int(before_balance) + int(total)), int(after_balance)
+        )
+        time.sleep(3)
+
+    def test_fund_community_pool_tx(self):
+        # fund community pool tx
+        status, fund_pool = tx_fund_communitypool("account1", amount_to_be_sent)
+        self.assertTrue(status)
+        self.assertTrue(fund_pool)
+
+        # query community pool fund
+        pool_fund = query_community_pool()[1]["pool"][0]["amount"]
+        time.sleep(5)
+
+        # fund community pool tx
+        status, fund_pool = tx_fund_communitypool("account1", amount_to_be_sent)
+        self.assertTrue(status)
+        self.assertTrue(fund_pool)
+        time.sleep(5)
+
+        after_pool_fund = query_community_pool()[1]["pool"][0]["amount"]
+        self.assertLess(float(pool_fund), float(after_pool_fund))
+        time.sleep(3)
+
+    def test_set_withdraw_address_tx(self):
+        # tx set withdraw address
+        status, set_addr = tx_set_withdraw_addr("account1", delegator2)
+        self.assertTrue(status)
+        self.assertTrue(set_addr)
+
+        # query balance
+        before_balance = query_balances(delegator2)[1]["balances"][0]["amount"]
+        # query rewards
+        before_reward = query_rewards_singleval(delegator1, validator1)[1]["rewards"][
+            0
+        ]["amount"]
+
+        # tx withdraw rewards
+        status, rewards = tx_withdraw_rewards("account2", validator1)
+        self.assertTrue(status)
+        self.assertTrue(rewards)
+
+        # query balance
+        after_balance = query_balances(delegator2)[1]["balances"][0]["amount"]
+        assert int(after_balance) == int(before_balance) + int(
+            float(before_reward)
+        ), "Set withdraw rewards tx failed"
```

Comment:
Can you just use `assertEqual`? Also, we should parse `tx_withdraw_rewards("account2", validator1)` response to get actual withdrawn-amount. It will be inside `logs` in the response